client/state/reader/follows/reducer.js

```diff
@@ -11,11 +11,95 @@ import {
 	READER_FOLLOW,
 	READER_UNFOLLOW,
 	READER_FOLLOWS_RECEIVE,
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_SUBSCRIBE_TO_NEW_COMMENT_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_COMMENT_EMAIL,
 } from 'state/action-types';
 import { prepareComparableUrl } from './utils';
 import { createReducer } from 'state/utils';
 
 /**
+ * Update a post subscription
+ *
+ * @param {object} state The current state from the items reducer
+ * @param {object} payload the payload from the subscribe / update / unsubscribe action
+ * @param {function} isEqualToCurrentState A function that determines if the payload will actually change the current state.
+ * Returns true if the state would not change.
+ * @param {object} newProps The new props to set on the delivery_methods.email object if the state will change
+ * @returns {object} the next state
+ */
+function updatePostSubscription( state, payload, isEqualToCurrentState, newProps ) {
```

Comment:
what I had in mind was that you could keep passing along the action instead of extracting the payload and do something like:
```js
const prevFollow = find( state, { blog_ID, payload.blogId } );
const nextFollow = { 
   ...prevFollow,
   delivery_methods: {
     email: { 
       send_posts: prevFollow.deliver_methods.email.send_post || ( action.type === READER_SUBSCRIBE_TO_NEW_POST_EMAIL ),
       post_delivery_frequency: prevFollow.delivery_method.email.post_delivery_frequency || ...,
     } 
   }
}

return isEqual( prevFollow, nextFollow ) 
   ? state
   : {
      ...state,
      [ prepareComparableUrl( follow.url ) ]: newFollow
    }  
```